### ISSUE TITLE
RuntimeError: dimension specified as 0 but tensor has no dimensions

### ISSUE THREAD

**Issue by tingkai-zhang** [allenai/allennlp#2201]
Hi, I got the error after i change the `cuda_device` in `Trainer` from an int to a list of int.

```
  File "/media/F/tingkai/anaconda3/envs/torch/lib/python3.7/site-packages/torch/nn/modules/module.py", line 477ll__
    result = self.forward(*input, **kwargs)
  File "/media/F/tingkai/anaconda3/envs/torch/lib/python3.7/site-packages/torch/nn/modules/loss.py", line 862, d
    ignore_index=self.ignore_index, reduction=self.reduction)
  File "/media/F/tingkai/anaconda3/envs/torch/lib/python3.7/site-packages/torch/nn/functional.py", line 1550, intropy
    return nll_loss(log_softmax(input, 1), target, weight, None, ignore_index, None, reduction)
  File "/media/F/tingkai/anaconda3/envs/torch/lib/python3.7/site-packages/torch/nn/functional.py", line 1403, is
    if input.size(0) != target.size(0):
RuntimeError: dimension specified as 0 but tensor has no dimensions
```

It works fine when i assign the model to only one gpu with anything else unchanged.
This problem happens at the end of one epoch training. That is, it occurs when the model processing the last batch of an epoch.
It was calculating the loss of last batch and i check that the label i passed to the model forward function turns out to be None (but the classify logits of that batch is correct ).

**Comment by matt-gardner** [comment]
@brendan-ai2, this seems like it would also be solved by switching to one batch per GPU (#2200), though we'd have to be careful about what to do when you get to the end of an epoch and there aren't as many batches left as there are GPUs.

**Comment by suki7** [comment]
I met a problem like yours, how to solve the problem ?
